larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -226,6 +230,34 @@ void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const
     {
         PandoraApi::Geometry::LineGap::Parameters parameters;
 
+        if (isDualPhase)
+        {
+            const bool  isGapInU ((std::fabs(gap.GetY2() - gap.GetY1()) > gap.GetMaxGapSize())); //Could have chosen Z here, resulting in switching Y<->Z and U<->V in the try{...} block below
+
+            try
+            {
+                parameters.m_lineGapType = ( isGapInU ? pandora::TPC_WIRE_GAP_VIEW_U : pandora::TPC_WIRE_GAP_VIEW_V); //If gapSizeY is too large then the gap is in Z, therefore should be in kU (i.e. kZ)
+                parameters.m_lineStartX = gap.GetX2();
+                parameters.m_lineEndX = gap.GetX1();
+                parameters.m_lineStartZ = ( isGapInU ? gap.GetZ1() : gap.GetY1());
+                parameters.m_lineEndZ = ( isGapInU ? gap.GetZ2() : gap.GetY2());
+            }
+            catch (const pandora::StatusCodeException &)
+            {
+                mf::LogWarning("LArPandora") << "CreatePandoraDetectorGaps - invalid line gap parameter provided, all assigned values must be finite, line gap omitted " << std::endl;
+                continue;
+            }
```

Comment:
None of the operations in these lines will throw an exception.  Please remove the `try/catch` block.

Reply:
Hi @knoepfel, the [lhs](https://github.com/PandoraPFA/PandoraSDK/blob/master/include/Pandora/ObjectCreation.h#L265-L269) parameter member variables here are actually all of type pandora::InputType, so a pandora::StatusCodeException will be raised if an attempt is made to assign a NaN or INF.

(The same structures are used to collect parameters for pandora::CaloHit creation, and this approach highlighted a number of issues in early uboone hit creation. Back then, we didn't have the try/catch blocks around the hit creation, so the job was terminated upon NaN assignment, and we were asked to update to just gently skip over any problem hits.)

Reply:
Thanks for the explanation, John.